### PR TITLE
chore: Support computability override in schema generation config

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-279922

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [ ] If changes include deprecations or removals I have added appropriate changelog entries.
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: tools/codegen/codespec/api_to_provider_spec_mapper_test.go [hunk 1]
```diff
@@ -24,327 +24,318 @@ type convertToSpecTestCase struct {
...
 	}
-
-	for name, tc := range testCases {
-		t.Run(name, func(t *testing.T) {
-			result, err := codespec.ToCodeSpecModel(tc.inputOpenAPISpecPath, tc.inputConfigPath, &tc.inputResourceName)
-			require.NoError(t, err)
-			assert.Equal(t, tc.expectedResult, result, "Expected result to match the specified structure")
-		})
-	}
+	runTestCase(t, tc)
```
Comment (AgustinBettati) [inline tools/codegen/codespec/api_to_provider_spec_mapper_test.go:96]: extracted common code that was define in each test

**Review by lantoli — APPROVED** [review]

File: tools/codegen/codespec/api_to_provider_spec_mapper_test.go [hunk 2]
```diff
@@ -24,327 +24,318 @@ type convertToSpecTestCase struct {
 }
 
 func TestConvertToProviderSpec(t *testing.T) {
-	testCases := map[string]convertToSpecTestCase{
-		"Valid input": {
-			inputOpenAPISpecPath: "testdata/api-spec.yml",
-			inputConfigPath:      "testdata/config-no-schema-opts.yml",
-			inputResourceName:    "test_resource",
+	tc := convertToSpecTestCase{
+		inputOpenAPISpecPath: "testdata/api-spec.yml",
```
Comment (EspenAlbert) [inline tools/codegen/codespec/api_to_provider_spec_mapper_test.go]: [nit] will there ever be more than 1 api-spec? Maybe a global constant

File: tools/codegen/codespec/config.go [hunk 3]
```diff
@@ -80,8 +80,23 @@ func applyAlias(attr *Attribute, attrPathName *string, schemaOptions config.Sche
...
+	}
+}
+
+func getComputabilityFromConfig(computability config.Computability) ComputedOptionalRequired {
+	if computability.Computed && computability.Optional {
+		return ComputedOptional
+	}
+	if computability.Optional {
+		return Optional
 	}
```
Comment (EspenAlbert) [inline tools/codegen/codespec/config.go:101]: what about computed only?

**Review by EspenAlbert — APPROVED** [review]
Two minor comments

File: tools/codegen/codespec/api_to_provider_spec_mapper_test.go (hunk 2 above)
Comment (AgustinBettati) [reply tools/codegen/codespec/api_to_provider_spec_mapper_test.go]: agree, extracted

File: tools/codegen/codespec/config.go (hunk 3 above)
Comment (AgustinBettati) [reply tools/codegen/codespec/config.go:101]: thanks, added this missing case

**Pull request merged**